File: week11/week11-2.py
# week11-2.py N! 函式呼叫函式,算N!
# 給 Python Tutor 執行
def func(n):
  if n == 1: return 1
  return n * func(n-1)
n = 4 # = int(input('請輸入1個數字:'))
ans = func(n)
print('答案是:', ans )

File: week11/week11-2.cpp
#include <stdio.h>
int func(int n) {
    if(n==1) return 1;
    return n * func(n-1);
}
int main()
{
    printf("請輸入1個數字: 4");
    int n=4;
    //scanf("%d", &n);
    int ans = func(n);
    printf("答案是:%d", ans);
}